packages/devtools/devtools/src/panels/mesh/NetworkPanel.tsx

```diff
@@ -0,0 +1,164 @@
+import { Toolbar } from "@dxos/aurora";
+import { mx } from "@dxos/aurora-theme";
+import { SVG, SVGContextProvider, createSvgContext } from '@dxos/gem-core';
+import { Graph, GraphData, GraphForceProjector, GraphLayoutNode, GraphLink, GraphModel, emptyGraph } from "@dxos/gem-spore";
+import { PeerState } from "@dxos/protocols/proto/dxos/mesh/presence";
+import { SpaceMember, useMembers } from "@dxos/react-client/echo";
+import { useIdentity } from "@dxos/react-client/halo";
+import { defaultMap } from "@dxos/util";
+import React, { useEffect, useMemo, useState } from 'react';
+import { PanelContainer } from "../../components";
+import { SpaceSelector } from "../../containers";
+import { useDevtoolsState } from "../../hooks";
+
+const NetworkPanel = () => {
+  const { space } = useDevtoolsState();
+  const identity = useIdentity();
+
+  const isMe = (node: NetworkGraphNode) => identity ? node.member?.identity.identityKey.equals(identity.identityKey) : false;
+
+  const members = useMembers(space?.key);
+  const [model] = useState(() => new NetworkGraphModel());
+  useEffect(() => {
+    if (members) {
+      model.setFromMemberList(members);
+      console.log(model.graph)
+    }
+  }, [members])
+
+  const context = createSvgContext();
+  const projector = useMemo(
+    () =>
+      new GraphForceProjector<NetworkGraphNode>(context, {
+        guides: true,
+        forces: {
+          manyBody: {
+            strength: -160,
+          },
+          link: {
+            distance: 120,
+            iterations: 5,
+          },
+          radial: {
+            radius: 60,
+            strength: 0.2,
+          },
+        },
+        attributes: {
+          radius: (node: GraphLayoutNode<NetworkGraphNode>, count) => isMe(node.data!) ? 10 : 6,
+        },
+      }),
+    [],
+  );
+
+  // TODO(dmaretskyi): Fix colors.
+  // TODO(dmaretskyi): Highlight our direct connections.
+  // TODO(dmaretskyi): Visualize data flowing: line thickness, running ticks, text stats.
+  // TODO(dmaretskyi): Show connections that are forming.
+  return (
+    <PanelContainer
+      toolbar={
+        <Toolbar.Root>
+          <SpaceSelector />
+        </Toolbar.Root>
+      }
+      className='overflow-auto'
+    >
+      <SVGContextProvider context={context}>
+        <SVG>
+          {/* <Markers /> */}
+          {/* <Grid /> */}
+          {/* <Zoom extent={[1 / 2, 2]}> */}
+          <Graph
+            model={model}
+            drag
+            arrows
+            projector={projector}
+            labels={{
+              text: (node: GraphLayoutNode<NetworkGraphNode>, highlight) => {
+                // TODO(dmaretskyi): CSS margins?
+                const margin = '       '
```

Comment:
doesn't the projector radius solve this?

Reply:
not really, the circle gets bigger and occludes the text